File: golang/674.longest-continuous-increasing-subsequence.go
func findLengthOfLCIS(nums []int) int {
	maxLength := 1
	length := 1
	n := len(nums)

	for i := 1; i < n; i++ {
		if nums[i-1] < nums[i] {
			length++
		} else {
			length = 1
		}

		maxLength = max(maxLength, length)
	}

	return maxLength
}

File: python/674.longest-continuous-increasing-subsequence.py
class Solution:
    def findLengthOfLCIS(self, nums: list[int]) -> int:

        n = len(nums)
        length, max_length = 1, 1

        for i in range(1, n):
            if nums[i - 1] < nums[i]:
                length += 1
            else:
                length = 1

            max_length = max(max_length, length)

        return max_length

